scripts/build-docker.sh

```diff
@@ -51,6 +51,9 @@ echo "Building ${GITUSER}/${GITREPO}:${TAG} docker image ..."
 docker build ${NOCACHE_FLAG} --pull -f ./docker/Dockerfile \
     --build-arg PROFILE="$PROFILE" \
     --build-arg BUILD_ARGS="$ARGS" \
+    --build-arg HTTP_PROXY="${HTTP_PROXY//localhost/host.docker.internal}" \
+    --add-host=host.docker.internal:host-gateway \
+    --network host \
```

Comment:
I had got the same concern and tested. It would stop working without `--add-host=host.docker.internal:host-gateway`.

Reply:
The following worked for me locally:

1. start a simple http server locally at 8000

2. create a Dockerfile:
```
FROM ubuntu:22.04
RUN apt-get update && apt-get install -yq curl
RUN env | grep -i _PROXY || true
RUN curl -I host.docker.internal:8000
```

3. build the docker:
```
docker build --no-cache --progress=plain --network host -t tmp .
```

It will show `curl` succeeds:

```
#7 0.072 HTTP/1.0 200 OK
#7 0.072 Server: SimpleHTTP/0.6 Python/3.11.6
#7 0.072 Date: Fri, 13 Oct 2023 08:18:42 GMT
#7 0.072 Content-type: text/html; charset=utf-8
#7 0.072 Content-Length: 474
```

Is that also the case from your side?